### PR TITLE
refactor(core): Simplify `OrchestrationService` (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
This PR merges `SingleModeSetup` into `OrchestrationService` and injects `MultiMainSetup` into it. This simplifies usage of `OrchestrationService` and allows multi-main mode to support workers view.

### CODE REVIEW

File: packages/cli/src/services/orchestration.service.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	async init() {
+		if (this.isInitialized) return;
+
+		await this.initPublisher();
+
+		if (this.isMultiMainSetupEnabled) await this.multiMainSetup.init();
+
+		this.isInitialized = true;
```
Comment (krynble) [inline packages/cli/src/services/orchestration.service.ts:64]: If multi main is not enable, should we also set this instance as `leader`?

File: packages/cli/src/commands/start.ts [hunk 2]
```diff
@@ -214,30 +213,21 @@ export class Start extends BaseCommand {
...
 
-		const multiMainSetup = Container.get(MultiMainSetup);
+		await orchestrationService.init();
 
-		await multiMainSetup.init();
+		await Container.get(OrchestrationHandlerMainService).init();
 
-		multiMainSetup.on('leadershipChange', async () => {
-			if (multiMainSetup.isLeader) {
+		orchestrationService.multiMainSetup.addListener('leadershipChange', async () => {
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/commands/start.ts:229]: This creates a cyclic dependency with `ActiveWorkflowRunner`.

**Comment by ivov** [comment]
@krynble Gave it another spin and it seems to keep working fine.

File: packages/cli/src/ActiveWorkflowRunner.ts [hunk 3]
```diff
@@ -634,6 +632,14 @@ export class ActiveWorkflowRunner {
 
 			const triggerCount = this.countTriggers(workflow, additionalData);
 			await this.workflowRepository.updateWorkflowTriggerCount(workflow.id, triggerCount);
+
+			if (shouldAddWebhooks || shouldAddTriggersAndPollers) {
+				this.logger.verbose(`Successfully started workflow ${dbWorkflow.display()}`, {
+					workflowName: dbWorkflow.name,
+					workflowId: dbWorkflow.id,
+				});
+				this.logger.info('     => Started');
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/ActiveWorkflowRunner.ts]: Thank you, please see [44a31b5](https://github.com/n8n-io/n8n/pull/8364/commits/44a31b55cb6a37d0da3b9d3aa6b24573c7f2dc03)

**Review by krynble — DISMISSED** [review]

**Comment by ivov** [comment]
@krynble Can you reapprove when you get the chance? Did some minor cleanup after #8401.

**Review by krynble — APPROVED** [review]

**Pull request merged**